plugin-build/src/main/kotlin/io/sentry/android/gradle/tasks/SentryCliExecTask.kt

```diff
@@ -0,0 +1,132 @@
+package io.sentry.android.gradle.tasks
+
+import io.sentry.android.gradle.SentryCliProvider
+import io.sentry.android.gradle.telemetry.SentryTelemetryService
+import io.sentry.android.gradle.util.info
+import java.io.File
+import org.apache.tools.ant.taskdefs.condition.Os
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Exec
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.Optional
+
+abstract class SentryCliExecTask : Exec() {
+
+    @get:Input
+    @get:Optional
+    abstract val debug: Property<Boolean>
+
+    @get:Input
+    abstract val cliExecutable: Property<String>
+
+    @get:Input
+    abstract val buildDirectory: Property<File>
```

Comment:
`m`: I think you don't even have to pass it as a task input, you can actually access it within the task itself: `project.layout.buildDirectory`